study-group/Google-Toast-Implementations.md

```diff
@@ -0,0 +1,101 @@
+# Google Toast Implementations
+This document studies the various ways Google implements toasts, in Polymer, Material UI, and Android.
+
+## Background
+Google has a range of different component library solutions which cover a variety of usages, many of which implement some toast-like element.
+The company offers:
+
+- [Polymer `<paper-toast>`](#polymer-paper-toast)
+- [Material UI Snackbar](#material-ui-snackbar)
+- [Android Toast](#android-toast)
+- [Android Snackbar](#android-snackbar)
+
+Interestingly, Google is the only library provider I could find who uses the term `snackbar` to refer to an actionable, dismissable toast.
+The implementations have internal consistency in design and implementation, and demonstrate popular patterns for toast APIs, which make them good candidates to examine.
+
+This study covers the details and trade-offs of the various toast-like feedback pop-ups Google libraries provide, by taking an in-depth look at each implementation, then extracting some key takeaways.
+
+## [Polymer `<paper-toast>`](https://github.com/PolymerElements/paper-toast)
+Polymer `<paper-toast>` is a library released by Google's Polymer team to provide a simple Material Design toast notification.
+
+### Design & Sample Code
+To use `<paper-toast>`, use something like this in HTML:
+
+```html
+<paper-toast text="Hello world!" opened></paper-toast>
+```
+
+This will display a notification with the text "Hello World!" that will stay open for 3 seconds (default).
+To change the default length of time it displays, set the `duration` attribute.
+
+```html
+<paper-toast text="This message closes in 5 seconds!" duration="5000" opened></paper-toast>
+```
+
+To control when the toast opens and closes, use the `.open()` and `.close()` methods from JavaScript - or use buttons to call those methods.
+For the toast to persist until `.close()` is called, set `duration` to 0.
+
+```html
+<button onclick="toast1.open()">Persistent toast</button>
+
+<paper-toast id="toast1" duration="0" text="This toast sticks around!">
+  <button onclick="toast1.toggle()">Close it!</button>
+</paper-toast>
+```
+
+To put a `<paper-toast>` into a specific container, set the `fitInto` attribute to the id of that container.
+
+```html
+<div id="container0">
+...
+</div>
+
+<paper-toast id="toast2" fitInto="container0" text="This fits perfectly!"></paper-toast>
+```
+
+Finally, to style the toast set the CSS properties `--paper-toast-background-color` and `--paper-toast-color` on the `id` of the toast in the stylesheet.
```

Comment:
Is this an exhaustive list? If so it might be good to mention that.

Reply:
Changed language to make clearer what we know, this is all they give in the docs and I can't really take the time to download it and test it other styling.